packages/mermaid/src/diagrams/class/classDb.ts

```diff
@@ -24,713 +24,720 @@ import type {
-    members.forEach((member) => addMember(className, member));
+  public setClassLabel(_id: string, label: string) {
+    const id = common.sanitizeText(_id, getConfig());
+    if (label) {
+      label = this.sanitizeText(label);
+    }
+
+    const { className } = this.splitClassNameAndType(id);
+    this.classes.get(className)!.label = label;
+    this.classes.get(className)!.text =
+      `${label}${this.classes.get(className)!.type ? `<${this.classes.get(className)!.type}>` : ''}`;
   }
-};
 
-export const addNote = function (text: string, className: string) {
-  const note = {
-    id: `note${notes.length}`,
-    class: className,
-    text: text,
-  };
-  notes.push(note);
-};
-
-export const cleanupLabel = function (label: string) {
-  if (label.startsWith(':')) {
-    label = label.substring(1);
-  }
-  return sanitizeText(label.trim());
-};
-
-/**
- * Called by parser when assigning cssClass to a class
- *
- * @param ids - Comma separated list of ids
- * @param className - Class to add
- */
-export const setCssClass = function (ids: string, className: string) {
-  ids.split(',').forEach(function (_id) {
-    let id = _id;
-    if (/\d/.exec(_id[0])) {
-      id = MERMAID_DOM_ID_PREFIX + id;
+  /**
+   * Function called by parser when a node definition has been found.
+   *
+   * @param id - Id of the class to add
+   * @public
+   */
+  public addClass(_id: string) {
+    const id = common.sanitizeText(_id, getConfig());
+    const { className, type } = this.splitClassNameAndType(id);
+    // Only add class if not exists
+    if (this.classes.has(className)) {
+      return;
+    }
+    // alert('Adding class: ' + className);
+    const name = common.sanitizeText(className, getConfig());
+    // alert('Adding class after: ' + name);
+    this.classes.set(name, {
+      id: name,
+      type: type,
+      label: name,
+      text: `${name}${type ? `&lt;${type}&gt;` : ''}`,
+      shape: 'classBox',
+      cssClasses: 'default',
+      methods: [],
+      members: [],
+      annotations: [],
+      styles: [],
+      domId: MERMAID_DOM_ID_PREFIX + name + '-' + classCounter,
+    } as ClassNode);
+
+    classCounter++;
+  }
+
+  private addInterface(label: string, classId: string) {
+    const classInterface: Interface = {
+      id: `interface${this.interfaces.length}`,
+      label,
+      classId,
+    };
+
+    this.interfaces.push(classInterface);
+  }
+
+  /**
+   * Function to lookup domId from id in the graph definition.
+   *
+   * @param id - class ID to lookup
+   * @public
+   */
+  public lookUpDomId(_id: string): string {
+    const id = common.sanitizeText(_id, getConfig());
+    if (this.classes.has(id)) {
+      return this.classes.get(id)!.domId;
+    }
+    throw new Error('Class not found: ' + id);
+  }
+
+  public clear() {
+    this.relations = [];
+    this.classes = new Map();
+    this.notes = [];
+    this.interfaces = [];
+    this.functions = [];
+    this.functions.push(this.setupToolTips);
```

Comment:
Can we turn `setupToolTips` into a normal class method and use `.bind(this);` here?

```suggestion
    this.functions.push(this.setupToolTips.bind(this));
```

See also https://github.com/mermaid-js/mermaid/blob/92efc24283716acd5147bdf1fb00c0997fca4c0f/packages/mermaid/src/diagrams/class/classDb.ts#L49

Reply:
Fixed in https://github.com/mermaid-js/mermaid/pull/6212/commits/b09f2e836af33a5e79082a0b72c57889a6ddeaf3